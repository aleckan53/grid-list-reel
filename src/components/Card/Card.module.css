.card {
  background: red;
}
/*# sourceMappingURL=Card.module.css.map */