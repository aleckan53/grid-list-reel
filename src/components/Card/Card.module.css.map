{
    "version": 3,
    "mappings": "AAGA,AAAA,KAAK,CAAC;EACJ,UAAU,EAAE,IAAI;EAChB,UAAU,EAAE,KAAK;EACjB,MAAM,EALC,aAAa,CADZ,cAAc;EAQtB,OAAO,EAAE,IAAI;EACb,qBAAqB,EAAE,YAAY;EACnC,kBAAkB,EAAE,WAAW;EAC/B,eAAe,EAAE,IAAI;EACrB,YAAY,EAAE,KAAK;EACnB,OAAO,EAAE,IAAI;EACb,mBAAmB,EAAE,0DAEY;CA0BlC;;AAvCD,AAeE,KAfG,CAeH,IAAI,CAAC;EACH,SAAS,EAAE,GAAG;EACd,MAAM,EAnBD,aAAa,CADZ,cAAc;CAqBrB;;AAlBH,AAoBE,KApBG,CAoBH,KAAK,CAAC;EACJ,SAAS,EAAE,IAAI;CAChB;;AAtBH,AAwBE,KAxBG,CAwBH,SAAS,CAAC;EACR,SAAS,EAAE,QAAQ;CACpB;;AA1BH,AA4BE,KA5BG,CA4BH,QAAQ,CAAC;EACP,SAAS,EAAE,OAAO;EAClB,MAAM,EAAE,KAAK;EACb,KAAK,EAAE,IAAI;EACX,OAAO,EAAE,IAAI;CACd;;AAjCH,AAmCE,KAnCG,CAmCH,OAAO,CAAC;EACN,SAAS,EAAE,MAAM;EACjB,UAAU,EAAE,MAAM;CACnB",
    "sources": [
        "Card.module.scss"
    ],
    "names": [],
    "file": "Card.module.css"
}