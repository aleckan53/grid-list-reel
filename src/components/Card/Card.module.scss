$black30: var(--black30);
$black50: var(--black50);
$border: var(--border);
$border-radius: var(--border-radius);
$shadow: var(--shadow);

.card {
  list-style: none;
  background: white;
  border: $border $black30;
  border-radius: $border-radius;
  box-shadow: $shadow $black30;

  display: grid;
  grid-template-columns: 25% auto 25%;
  grid-template-rows: 80px 0% auto;
  grid-column-gap: 1rem;
  grid-row-gap: 1rem;
  padding: 1rem;
  grid-template-areas: "img info progress"
                       "details details details"
                       "expand expand expand";

  &.expand {
    grid-template-rows: 80px auto auto;
  }

  .img {
    grid-area: img;
    border: $border $black30;
    border-radius: 10px;
    background-size: contain;
    background-position: center;
    background-repeat: no-repeat;
    position: relative;
  }

  .info {
    grid-area: info;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    font-size: .8rem;

    h3 {
      font-size: 5vw;
    }
  }

  .progress {
    grid-area: progress;
    display: grid;
    place-content: center;
    font-size: 2rem;
  }

  .details {
    grid-area: details;
    width: 100%;
    display: none;
    margin-top: .5rem;

    &.expand {
      display: block;
    }

    p {
      display: flex;
      justify-content: space-between;
      margin-bottom: 1rem;
    }

    button {
      width: 45%;
      height: 30px;
      border: $border $black30;
      background: transparent;
      border-radius: 10px;
      &:nth-of-type(2) {
        margin: .5rem 0 1rem 10%;
        color: gray;
      }
    }
  }

  .toggle {
    grid-area: expand;
    display: block;
    width: 100%;
    background: transparent;
    border: none;
    position: relative;
    outline: none;

    &::after {
      content: '';
      width: 100%;
      height: 1px;
      background: $black30;
      position: absolute;
      left: 0;
      top: -1rem;
    }
}
}