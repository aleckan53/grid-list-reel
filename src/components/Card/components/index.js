import Image from './Image';
import Info from './Info';
import Progress from './Progress';
import Details from './Details';

export {
  Image,
  Info,
  Progress,
  Details
};