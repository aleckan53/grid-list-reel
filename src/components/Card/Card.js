import React from 'react';
import styles from './css/Card.module.css';
import { Image, Info, Progress, Details } from './components';

import { useToggle } from 'hooks';

const Card = ({ data }) => {

  const { "data-toggle": viewMore, onClick: toggleDropDown } = useToggle();

  return (
    <li className={`${styles.card } ${viewMore && styles.expand}`}>
      <Image className={styles.img} img={data.img}/>
      <Info className={styles.info} data={data}/>
      <Progress className={styles.progress} percent={data.percentSaved}/>
      <Details className={`${styles.details } ${viewMore && styles.expand}`} data={data}/>
      <button 
        onClick={toggleDropDown}
        className={ styles.toggle }>
        {viewMore ? 'Hide details' : 'View details'}
      </button>
    </li>
  );
};

export default Card;