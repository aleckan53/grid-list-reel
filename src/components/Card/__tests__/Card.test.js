import React from 'react';
import ReactDOM from 'react-dom';

import Card from '../Card';

describe('Card component', () => {
  it('matches the snapshot', () => {
    const data =   {
      img: 'URL',
      name: 'Name',
      status: 'active',
      total: 35000,
      started: '2017-5-1',
      perDay: 15,
      saved: 12000,
      percentSaved: 34,
    };
    const root = document.createElement('div');
    ReactDOM.render(<Card data={data}/>, root);
    expect(root).toMatchSnapshot();
    ReactDOM.unmountComponentAtNode(root);
  });
});