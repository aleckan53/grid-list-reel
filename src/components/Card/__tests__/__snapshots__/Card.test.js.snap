// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component matches the snapshot 1`] = `
<div>
  <li
    class="card false"
  >
    <div
      class="img"
      style="background-image: url(URL);"
    />
    <div
      class="info"
    >
      <h3>
        Name
      </h3>
      <p>
        $
        12k
         of $
        35k
      </p>
      <p
        data-status="active"
      >
        active
      </p>
    </div>
    <div
      class="progress"
    >
      <span>
        34
      </span>
      <svg
        viewBox="0 0 100 100"
      >
        <circle
          cx="50%"
          cy="50%"
          data-circle="bottom"
          fill="transparent"
          r="45"
          stroke-width="7"
        />
        <circle
          cx="50%"
          cy="50%"
          data-circle="top"
          data-status="active"
          fill="transparent"
          id="circle"
          r="45"
          stroke-width="7"
          style="stroke-dashoffset: 0;"
        />
      </svg>
    </div>
    <div
      class="details false"
    >
      <p>
        <span>
          Days left:
        </span>
        <span>
          12
        </span>
      </p>
      <p>
        <span>
          $ per day:
        </span>
        <span>
          $
          15
        </span>
      </p>
      <p>
        <span>
          Start date:
        </span>
        <span>
          2017-5-1
        </span>
      </p>
      <p>
        <span>
          Estimated finsh date:
        </span>
        <span>
          2017-5-1
        </span>
      </p>
      <div
        class="controls"
      >
        <button>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M32 384h272v32H32zM400 384h80v32h-80zM384 447.5c0 17.949-14.327 32.5-32 32.5-17.673 0-32-14.551-32-32.5v-95c0-17.949 14.327-32.5 32-32.5 17.673 0 32 14.551 32 32.5v95z"
            />
            <g>
              <path
                d="M32 240h80v32H32zM208 240h272v32H208zM192 303.5c0 17.949-14.327 32.5-32 32.5-17.673 0-32-14.551-32-32.5v-95c0-17.949 14.327-32.5 32-32.5 17.673 0 32 14.551 32 32.5v95z"
              />
            </g>
            <g>
              <path
                d="M32 96h272v32H32zM400 96h80v32h-80zM384 159.5c0 17.949-14.327 32.5-32 32.5-17.673 0-32-14.551-32-32.5v-95c0-17.949 14.327-32.5 32-32.5 17.673 0 32 14.551 32 32.5v95z"
              />
            </g>
          </svg>
          <span>
            Edit
          </span>
        </button>
        <button>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M96 448h106.7V64H96v384zM309.3 64v384H416V64H309.3z"
            />
          </svg>
          <span>
            Pause
          </span>
        </button>
      </div>
    </div>
    <button
      class="toggle"
    >
      View details
    </button>
  </li>
</div>
`;
